### PR TITLE
Feature/Checkbox rework from CC into FC

### PR DESCRIPTION
**Pull request by vik753**

### CODE REVIEW

File: uui-components/src/inputs/__tests__/Checkbox.test.tsx [hunk 1]
```diff
@@ -103,4 +120,14 @@ describe('Checkbox', () => {
         expect(onBlur).toHaveBeenCalled();
         expect(input).not.toHaveFocus();
     });
+
+    it('when state equals isInvalid: true, Checkbox_container must have a \'uui-invalid\' class', async () => {
```
Comment (AlekseyManetov) [inline uui-components/src/inputs/__tests__/Checkbox.test.tsx:124]: Such cases better check via snapshot tests, just render Checkbox with invalid state. I guess we can add this case to the snapshot in UUI.

File: uui/components/inputs/__tests__/Checkbox.test.tsx [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
+describe('TestComponent', () => {
+    it('should render with minimum props', async () => {
+        const tree = await renderSnapshotWithContextAsync(<Checkbox value={ true } onValueChange={ jest.fn } />);
         expect(tree).toMatchSnapshot();
     });
 
-    it('should be rendered correctly', () => {
-        const tree = renderer.create(<Checkbox value={ null } onValueChange={ jest.fn } size="18" mode="cell" />).toJSON();
+    it('should render with maximum props', async () => {
+        const tree = await renderSnapshotWithContextAsync(<Checkbox value={ null } onValueChange={ jest.fn } size="18" mode="cell" />);
```
Comment (AlekseyManetov) [inline uui/components/inputs/__tests__/Checkbox.test.tsx:12]: Add wider range of different props to this test

**Review by AlekseyManetov — CHANGES_REQUESTED** [review]

File: uui-components/src/inputs/__tests__/Checkbox.test.tsx (hunk 1 above)
Comment (vik753) [reply uui-components/src/inputs/__tests__/Checkbox.test.tsx:124]: done

File: uui/components/inputs/__tests__/Checkbox.test.tsx (hunk 2 above)
Comment (vik753) [reply uui/components/inputs/__tests__/Checkbox.test.tsx:12]: done

**Pull request merged**